src/doghouse/adapters/github/gh_cli_adapter.py

```
@@ -0,0 +1,190 @@
+            state = check.get("conclusion") or check.get("state")
+            check_name = check.get("context") or check.get("name")
+
+            if state in ["FAILURE", "ERROR", "CANCELLED", "ACTION_REQUIRED"]:
+                blockers.append(Blocker(
+                    id=f"check-{check_name}",
+                    type=BlockerType.FAILING_CHECK,
+                    message=f"Check failed: {check_name}",
+                    severity=BlockerSeverity.BLOCKER
+                ))
+            elif state in ["PENDING", "IN_PROGRESS", "QUEUED", None]:
+                if check.get("status") != "COMPLETED" or state in ["PENDING", "IN_PROGRESS"]:
```

Ruff (0.15.7)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)